conans/test/functional/command/search_test.py

```diff
@@ -1076,3 +1078,193 @@ class Test(ConanFile):
             client.run("search Test/0.1@lasote/testing  %s --outdated" % remote)
             self.assertIn("os: Windows", client.user_io.out)
             self.assertNotIn("os: Linux", client.user_io.out)
+
+
+@unittest.skipUnless(TestClient().revisions,
+                     "The test needs revisions activated, set CONAN_CLIENT_REVISIONS_ENABLED=1")
+class SearchRevisionsTest(unittest.TestCase):
+
+    def search_recipe_revisions_test(self):
+        test_server = TestServer(users={"user": "password"})  # exported users and passwords
+        servers = {"default": test_server}
+        client = TestClient(servers=servers, users={"default": [("user", "password")]})
+
+        conanfile = """
+from conans import ConanFile
+class Test(ConanFile):
+    pass
+"""
+        client.save({"conanfile.py": conanfile})
+        client.run("export . lib/1.0@user/testing")
+        client.run("upload lib/1.0@user/testing -c")
+
+        # List locally
+        client.run("search lib/1.0@user/testing --revisions")
+        self.assertIn("bd761686d5c57b31f4cd85fd0329751f (No time)", client.out)
+
+        # Create new revision and upload
+        client.save({"conanfile.py": conanfile + "# force new rev"})
+        client.run("export . lib/1.0@user/testing")
+
+        client.run("search lib/1.0@user/testing --revisions")
+        self.assertNotIn("bd761686d5c57b31f4cd85fd0329751f", client.out)
+        self.assertIn("a94417fca6b55779c3b158f2ff50c40a (No time)", client.out)
+
+        # List remote
+        client.run("upload lib/1.0@user/testing -c")
+        client.run("search lib/1.0@user/testing -r default --revisions")
+        self.assertIn("bd761686d5c57b31f4cd85fd0329751f", client.out)
+        self.assertIn("a94417fca6b55779c3b158f2ff50c40a", client.out)
+        self.assertNotIn("(No time)", client.out)
+        now = datetime.datetime.now()
+        # Checking at least the today's date appears in the test
+        today_str = "(%s" % datetime_to_str(now).split(" ")[0]
+        self.assertIn(today_str, client.out)
```

Comment:
Maybe you can mock the returning value of the function that the server is calling to get the timestamp. Then you can check the full line in the `self.assertIn("<recipe id> (<timestamp>)")` lines above

Reply:
I tried it. I assume my incapacity of making it work. If this is something easy for you I would love to see it working.

Reply:
Check it here: https://github.com/lasote/conan/pull/18

Reply:
Thanks so much. I don't know why was not working for me. Will cherrypick the changes and upgrade.

Reply:
Close the PR then and I will delete my branch. Thanks